Title: Implementation of "seturgent" Function
Author: Eric Pruitt (https://www.codevat.com/, https://github.com/ericpruitt/)

Adds a new function called "seturgent" for setting the X11 window urgency hint.

diff --git a/dwm.c b/dwm.c
index b2bc9bd..04a079c 100644
--- a/dwm.c
+++ b/dwm.c
@@ -204,6 +204,7 @@ static void setfullscreen(Client *c, int fullscreen);
 static void setlayout(const Arg *arg);
 static void setmfact(const Arg *arg);
 static void setup(void);
+static void seturgent(Client *c);
 static void showhide(Client *c);
 static void sigchld(int unused);
 static void spawn(const Arg *arg);
@@ -1603,6 +1604,18 @@ setup(void)
 }
 
 void
+seturgent(Client *c) {
+	XWMHints *wmh;
+
+	c->isurgent = 1;
+	if (!(wmh = XGetWMHints(dpy, c->win)))
+		return;
+	wmh->flags |= XUrgencyHint;
+	XSetWMHints(dpy, c->win, wmh);
+	XFree(wmh);
+}
+
+void
 showhide(Client *c)
 {
 	if (!c)
